lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -18,13 +18,16 @@ use vrl::value::{Kind, Value};
 
 use super::{default_lossy, Deserializer};
 use crate::gelf::GELF_TARGET_PATHS;
-use crate::{gelf_fields::*, VALID_FIELD_REGEX};
+use crate::{gelf_fields::*, VALID_DECODE_FIELD_REGEX};
 
 // On GELF decoding behavior:
 //   Graylog has a relaxed decoding. They are much more lenient than the spec would
 //   suggest. We've elected to take a more strict approach to maintain backwards compatibility
 //   in the event that we need to change the behavior to be more relaxed, so that prior versions
 //   of vector will still work with the new relaxed decoding.
+//
+//   Additionally, Graylog's own GELF Output produces GELF messages with any field names present
+//   in the sending Stream, exceeding the specified field name character set.
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ I'd just maybe expand/re-word this a little

Also, this comment should be added to the encoder now too.

```suggestion
//   Graylog's own GELF Output produces GELF messages with any field names present
//   in the sending Stream, violating it's own spec.
//   The GELF codec has relaxed parsing on the field names compared to the spec.
//   In particular, we allow the `@` symbol, which is used in Graylog's output messages. 
```